### PR TITLE
Use CollectionMoveModal in query builder

### LINKED ISSUE #7709: Moving saved questions should use new Move modal
It should use the modal that we're now using for moving dashboards instead of this one:

collections_move_modal

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #7709

### CODE REVIEW

**Comment by salsakran** [comment]
this looks a little off visually to me.
screen shot 2018-06-18 at 1 06 10 pm

**Comment by mazameli** [comment]
Yeah, I think at the very least we'd want to slap a max-width of ~550px on that. That's what the existing full-screen modal does:

screen shot 2018-06-18 at 1 41 21 pm

Another thing I'll add to my to-do list, though, is inventorying our use of full-screen vs. overlay modals. We've gotten ourselves into a weird place where it doesn't feel super predictable what kind of modal will be shown in any given part of the product.

File: frontend/src/metabase/query_builder/components/QueryHeader.jsx [hunk 1]
```diff
@@ -289,7 +289,6 @@ export default class QueryHeader extends Component {
 
         buttonSections.push([
           <ModalWithTrigger
-            ref="move"
             key="move"
             full
```
Comment (kdoh) [inline frontend/src/metabase/query_builder/components/QueryHeader.jsx]: We should remove the full here to keep the experience for moving items the same throughout the product.

**Review by kdoh — CHANGES_REQUESTED** [review]

**Comment by tlrobinson** [comment]
Fixed screenshot 2018-06-19 10 51 51

**Review by kdoh — APPROVED** [review]

**Comment by tlrobinson** [comment]
One thing that's not ideal is the scroll behavior of the `CollectionMoveModal`. Currently the whole modal scrolls but we probably want a fixed header/footer with just the list scrolling. I tried adding `scroll-y` to the `CollectionPicker` but that didn't work. @kdoh ideas?

**Pull request merged**